mantis-runtime/src/main/java/io/mantisrx/runtime/source/http/impl/ClientWithResponse.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.runtime.source.http.impl;
+
+import lombok.Value;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClient;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClientResponse;
+
+@Value
+class ClientWithResponse<R, E> {
```

Comment:
This is an interesting class to exist. Could you please add javadoc for why it's needed OR how it's used?

Reply:
I have moved this within HttpSourceImpl as an inner class because it's only used from within it. I think the JavaDoc is unnecessary, given that this is strictly an implementation detail.